src/Nixfmt/Util.hs

```diff
@@ -0,0 +1,52 @@
+module Nixfmt.Util
+    ( manyList
+    , someList
+    , manyP
+    , someP
+    , manyText
+    , someText
+    , commonPrefix
+    , dropCommonIndentation
+    ) where
+
+import           Data.Char
+import           Data.Maybe
+import           Data.Text       as Text hiding (map, takeWhile)
+import           Data.Void
+import           Text.Megaparsec hiding (empty)
+
+type Parser = Parsec Void Text
+
+someP :: (Char -> Bool) -> Parser Text
+someP = takeWhile1P Nothing
+
+manyP :: (Char -> Bool) -> Parser Text
+manyP = takeWhileP Nothing
+
+someText :: Parser Text -> Parser Text
+someText p = Text.concat <$> some p
+
+manyText :: Parser Text -> Parser Text
+manyText p = Text.concat <$> many p
+
+someList :: Parser [a] -> Parser [a]
+someList p = Prelude.concat <$> some p
+
+manyList :: Parser [a] -> Parser [a]
+manyList p = Prelude.concat <$> many p
```

Comment:
Overall, all of these utils have confusing names and semantics. What’s the value of having them, it doesn’t seem too hard to just inline their definitions, which will, in my opinion, result in better readability.